### DISCUSSION TITLE
PHP v 8.1.2

### DISCUSSION THREAD

**Discussion by craigbass76** [mpdf/mpdf#1966] [Q&A]
### Guidelines

- [X] I understand that [if I fail to adhere to contribution guidelines and/or fail to provide all required details, this issue may be closed without review](https://github.com/mpdf/mpdf/blob/development/.github/CONTRIBUTING.md).

### Description of the bug

I can't seem to install  to run on my local web server (which uses PHP 8.1.2)

Is there a way around this without me having to downgrade my version of PHP?

### mPDF version

8

### PHP Version and environment (server type, cli provider etc., enclosing libraries and their respective versions)

8.1.2

### Reproducible PHP+CSS+HTML snippet suffering by the error

`composer require mpdf/mpdf dev-php8-support` gives me errors, and I guess the important part is:
```
Package mpdf/mpdf at version dev-php8-support has requirements incompatible with your PHP version, PHP extensions and Composer version:
...~8.0.0 which does not match your installed version 8.1.2...
```

**Comment by finwe** [comment]
Run just `composer require mpdf/mpdf`, why would you install dev-php8-support?

**Comment by craigbass76** [comment]
With that command, I'm getting version 6 something, and running into the curly braces issue.

Sorry for sticking this is issues at first - I wasn't sure which way to go with it...

**Reply by finwe** [reply]
You can try clearing composer caches and removing composer.lock first. 

Or run `composer require mpdf/mpdf "^8.2"`

**Reply by finwe** [reply]
Also make sure you don't already have mpdf/mpdf with older version item in the composer.json

**Comment by craigbass76** [comment]
Thank you.
Is that command in a doc somewhere that I missed? I ask so I can dope slap myself if it was, and because now I'm having a `Uncaught Error: Class "mPDF" not found in...` error, but I know by reading some other docs and forum (Github and StackExchange) posts that calling mpdf has changed over time. I'm doing `$mpdf = new \mPDF();` but think I read it used to be some variation (`\mPDF\mPDF` was one I saw)

**Reply by finwe** [reply]
Which command?

The class name has changed, but the other way around. Class `\mPDF` is for version 6.x, in 7+ it's [namespaced and PSR-2 cased](https://github.com/mpdf/mpdf/releases/tag/v7.0.0) to `\Mpdf\Mpdf`.

**Comment by craigbass76** [comment]
Thanks @finwe- I'm making legit PDFs with proper filenames now. I've got another question/issue, but I'll start a different thread.
